frontend/src/App.jsx

```diff
@@ -15,35 +14,57 @@ const App = () => {
   // hold board cards in an array of board components
   const [boards, setBoards] = useState([]);
   const [modalOpen, setModalOpen] = useState(false);
+  const [searchedBoards, setSearchedBoards] = useState([]);
+  const [searchMode, setSearchMode] = useState(false);
+  // error state if no results from search
+  const [noResults, setNoResults] = useState(false);
+
+  useEffect(() => {
+    // when search mode or searched boards changes indicates action in search bar, may set no results to true
+    setNoResults(searchMode && searchedBoards.length === 0)
+  }, [searchMode, searchedBoards])
 
   const toggleModal = () => {
       setModalOpen((prev) => !prev);
   }
 
+  const handleSearch = (searchedList) => {
+    setSearchedBoards(searchedList);
+    setSearchMode(true);
+  }
+
+  const toggleMode = () => {
+    setSearchMode((prev) => !prev)
+  }
+
+  const handleNewBoard = (newBoard) => {
+    setBoards((prev) => [...prev, newBoard])
+  }
+
+
+
   return (
     <div className="app-container">
       <section id="banner">
         <h1>Kudos Board</h1>
       </section>
       <header className='homepage-header'>
         <div className='toolbar'>
-          <SearchBar />
+          <SearchBar boardList={boards} onSearch={handleSearch} searchMode={toggleMode}/>
           <div className="category-btns">
             {categoryOptions.map((category) => {
               return (<CategoryButton key={uuidv4()} category={category} />)
             })}
           </div>
           <button onClick={toggleModal} className="buttons">Create New Board</button>
-          {/* {modalOpen && <CreateBoard onCloseModal={toggleModal} onCreate={handleCreateBoard} />} */}
-          {modalOpen && <CreateBoard onCloseModal={toggleModal} onCreate={(newBoard) => {setBoards((prev) => [...prev, newBoard])}} />}
+          {modalOpen && <CreateBoard onCloseModal={toggleModal} onCreate={handleNewBoard} />}
         </div>
-        {/* <Toolbar onCreate/> */}
       </header>
       <main>
-        <BoardList boardList={boards} />
-
-        {/* <BoardPage boardInfo={boards[0]} /> */}
-        {/* <CreateCard/> */}
+        {
+          // if in search mode, present search boards, otherwise present list of boards
+          searchMode ? <BoardList boardList={searchedBoards} searchMode={searchMode} noResults={noResults}/> : <BoardList boardList={boards} searchMode={searchMode} noResults={noResults}/>
```

Comment:
This could probably be 
`<BoardList boardList={searchMode ? searchedBoards L boards} searchMode={searchMode} noResults={noResults}/>` instead?